File: lib/java/opentoken/src/main/java/com/truveta/opentoken/attributes/person/BirthYearAttribute.java
/**
 * Copyright (c) Truveta. All rights reserved.
 */
package com.truveta.opentoken.attributes.person;

import java.util.List;

import com.truveta.opentoken.attributes.general.YearAttribute;
import com.truveta.opentoken.attributes.validation.YearRangeValidator;

/**
 * Represents the birth year attribute.
 * 
 * This class extends YearAttribute and provides functionality for working with
 * birth year fields. It recognizes "BirthYear" as a valid alias for this
 * attribute type.
 * 
 * The attribute performs normalization on input values by trimming whitespace
 * and validates that the birth year is a 4-digit year between 1910 and the current year.
 */
public class BirthYearAttribute extends YearAttribute {

    private static final String NAME = "BirthYear";
    private static final String[] ALIASES = new String[] { NAME, "YearOfBirth" };

    public BirthYearAttribute() {
        super(List.of(new YearRangeValidator()));
    }

    @Override
    public String getName() {
        return NAME;
    }

    @Override
    public String[] getAliases() {
        return ALIASES;
    }

}

File: lib/python/opentoken/src/main/opentoken/attributes/person/birth_year_attribute.py
"""
Copyright (c) Truveta. All rights reserved.
"""

from typing import List
from opentoken.attributes.general.year_attribute import YearAttribute
from opentoken.attributes.validation.year_range_validator import YearRangeValidator


class BirthYearAttribute(YearAttribute):
    """Represents the birth year attribute.

    This class extends YearAttribute and provides functionality for working with
    birth year fields. It recognizes "BirthYear" as a valid alias for this attribute type.

    The attribute performs normalization on input values by trimming whitespace
    and validates that the birth year is a 4-digit year between 1910 and the current year.
    """

    NAME = "BirthYear"
    ALIASES = [NAME]

    def __init__(self):
        super().__init__(additional_validators=[YearRangeValidator()])

    def get_name(self) -> str:
        """Get the name of the attribute.

        Returns:
            str: The name "BirthYear"
        """
        return self.NAME

    def get_aliases(self) -> List[str]:
        """Get the aliases for the attribute.

        Returns:
            List[str]: A list containing the aliases for this attribute
        """
        return self.ALIASES.copy()
